### PR TITLE
secondaryFile conformance changes v1.0.2

### PR DESCRIPTION
**Pull request by tetron**
Change test usage to use basename and path (which have well defined filename semantics) instead of location (which may be anything.)

### CODE REVIEW

File: v1.0/v1.0/search.cwl [hunk 1]
```diff
@@ -33,9 +33,9 @@ $graph:
       secondaryFiles:
         - ".idx1"
         - "^.idx2"
-        - '$(self.location+".idx3")'
-        - '$({"location": self.location+".idx4", "class": "File"})'
-        - '${ return self.location+".idx5"; }'
+        - '$(self.basename).idx3'
+        - '${ return self.basename+".idx4"; }'
+        - '$({"path": self.path+".idx5", "class": "File"})'
```
Comment (mr-c) [inline v1.0/v1.0/search.cwl]:
```
Test failed: /var/lib/jenkins/jobs/cwl-spec-pr/workspace/implementation/cwltool/version/v1.0/cwltool-venv/bin/cwltool --enable-dev --outdir=/tmp/tmpYGZvn8 --quiet 'v1.0/search.cwl#main' v1.0/search-job.json
Test InitialWorkDirRequirement linking input files and capturing secondaryFiles
on input and output.

Returned non-zero
[job index] Job error:
v1.0/search.cwl:29:5: Error collecting output for parameter 'result':
v1.0/search.cwl:33:7:   Expression evaluation error:
v1.0/search.cwl:33:7:   self does not contain key 'basename'
[job index] completed permanentFail
[step index] Output is missing expected field file:///var/lib/jenkins/jobs/cwl-spec-pr/workspace/implementation/cwltool/version/v1.0/common-workflow-language/v1.0/v1.0/search.cwl#main/index/result
[step index] completed permanentFail
Final process status is permanentFail
```

**Comment by tetron** [comment]
cwltool fix at https://github.com/common-workflow-language/cwltool/pull/516

However cwltool changes behavior in a way that fails the conformance test in master (which is changed by this branch).  So we need to merge this first?

File: v1.0/Process.yml [hunk 2]
```diff
@@ -447,10 +447,21 @@ $graph:
...
 
         If the value is an expression, the value of `self` in the expression
-        must be the primary input or output File to which this binding applies.
-
-        If the value is a string, it specifies that the following pattern
-        should be applied to the `location` of the primary file:
+        must be the primary input or output File object to which this binding
+        applies.  The `basename` field must be present in `self`.  For
+        `CommandLineTool` outputs the `path` field must also be present.  The
+        expression must return a filename string relative to the path to the
```
Comment (mr-c) [inline v1.0/Process.yml]: Returning an array of strings isn't valid?

**Comment by mr-c** [comment]
Since this also causes Rabix Bunny to fail I'd like some feedback from @StarvingMarvin @simonovic86 and @mayacoda

**Comment by tetron** [comment]
Updated a bit more.  Bumped to v1.0.2.  Expanded conformance test with more secondaryFile cases.

**Pull request merged**